packages/wrangler/src/config/index.ts

```diff
@@ -30,7 +31,8 @@ export function readConfig<CommandArgs>(
 	configPath: string | undefined,
 	// Include command specific args as well as the wrangler global flags
 	args: CommandArgs &
-		Pick<OnlyCamelCase<CommonYargsOptions>, "experimentalJsonConfig">
+		Pick<OnlyCamelCase<CommonYargsOptions>, "experimentalJsonConfig">,
+	requirePagesConfig?: boolean
```

Comment:
And with this new parameter our journey to the darkside is complete as we have ended up leaking the pages specific stuff into the general wrangler config stuff. 😢

Reply:
Let's land this as-is since it does fix the problem for DevWeek. But I would love it if we could circle back and have another look at how to make this stuff a bit less convoluted.

Reply:
Yeah, I don't love this, but it fixes the problem for now. We should definitely revisit, but I think there might be some pieces here that we fundamentally can't pull apart, if we want to maintain a single `readConfig()` interface.